src/MicroHs/Deriving.hs

```diff
@@ -251,20 +251,36 @@ derBounded _ (c, _) _ e = cannotDerive "Bounded" c e
 --------------------------------------------
 
 derEnum :: Deriver
-derEnum mctx lhs cs@(_:_) enm | all isNullary cs = do
+derEnum mctx lhs cs@(c0:_) enm | all isNullary cs = do
   hdr <- mkHdr mctx lhs cs enm
   let loc = getSLoc enm
 
       mkFrom (Constr _ _ c _) i =
         eEqn [EVar c] $ ELit loc (LInt i)
       mkTo (Constr _ _ c _) i =
         eEqn [ELit loc (LInt i)] $ EVar c
+      eFirstCon = let Constr _ _ c _ = c0 in tCon c
+      eLastCon = let Constr _ _ c _ = last cs in tCon c
 
       iFromEnum = mkIdentSLoc loc "fromEnum"
       iToEnum = mkIdentSLoc loc "toEnum"
+      iEnumFrom = mkIdentSLoc loc "enumFrom"
+      iEnumFromThen = mkIdentSLoc loc "enumFromThen"
+      iEnumFromTo = mkIdentSLoc loc "enumFromTo"
+      iEnumFromThenTo = mkIdentSLoc loc "enumFromThenTo"
       fromEqns = zipWith mkFrom cs [0..]
       toEqns   = zipWith mkTo   cs [0..]
-      inst = Instance hdr [Fcn iFromEnum fromEqns, Fcn iToEnum toEqns]
+      enumFromEqn =
+        -- enumFrom x = enumFromTo x (last cs)
+        let x = EVar (mkIdentSLoc loc "x")
+        in eEqn [x] (eAppI2 iEnumFromTo x eLastCon)
```

Comment:
Does `enumFromTo` have to be qualified with `B@` here?

Reply:
Yes. Any global identifiers on the RHS should be qualified. (It's unlikely to go wrong here, but it could.)